src/terrama2/core/data-model/DataManager.cpp

```diff
@@ -104,21 +104,29 @@ void terrama2::core::DataManager::add(DataSeriesPtr dataseries)
       throw terrama2::InvalidArgumentException() << ErrorDescription(errMsg);
     }
 
-    auto itPr = providers_.find(dataseries->dataProviderId);
-    if(itPr == providers_.end())
+    try
+    {
+      providers_.at(dataseries->dataProviderId);
+    }
+    catch (const std::out_of_range&)
     {
       QString errMsg = QObject::tr("Can not add a dataseries with a non-registered data provider: %1.").arg(dataseries->dataProviderId);
       TERRAMA2_LOG_ERROR() << errMsg;
       throw terrama2::InvalidArgumentException() << ErrorDescription(errMsg);
     }
 
-    auto itDs = dataseries_.find(dataseries->id);
-    if(itDs != dataseries_.end())
+    try
     {
-      QString errMsg = QObject::tr("DataSeries already registered. Is this an update?");
+      dataseries_.at(dataseries->id);
+
+      QString errMsg = QObject::tr("DataSeries already registered.");
       TERRAMA2_LOG_ERROR() << errMsg;
       throw terrama2::InvalidArgumentException() << ErrorDescription(errMsg);
     }
+    catch (const std::out_of_range&)
+    {
+
```

Comment:
Why there is no error handling for this case? If not necessary leave a comment in the catch clause